src/database/iterator.rs

```diff
@@ -44,28 +44,14 @@ pub struct Iterator<'a, K: Key + 'a> {
 ///
 /// Returns just the keys.
 pub struct KeyIterator<'a, K: Key + 'a> {
-    start: bool,
-    // Iterator accesses the Database through a leveldb_iter_t pointer
-    // but needs to hold the reference for lifetime tracking
-    #[allow(dead_code)]
-    database: PhantomData<&'a Database<K>>,
-    iter: RawIterator,
-    from: Option<&'a K>,
-    to: Option<&'a K>,
+    inner: Iterator<'a, K>,
```

Comment:
I think you could write this as `pub struct KeyIterator<'a, K: Key + 'a>(Iterator<'a, K>);` and then use `self.0` instead of `self.inner`.

Reply:
Yes, I wasn't sure which to use in this context and I simply chose to keep the current form. I'm open to changing it if there's wide consensus that the other one is better.

Reply:
I'm generally not a fan of one-tuple structs just for brevity. `inner` is idiomatic in that case.